### PR TITLE
Explain analyze will detail memory used across slices.

### PR DESCRIPTION
**Pull request by sambitesh**
We will be adding an extra line detailing total memory used across slices to explain analyze. This will help us in profiling memory usage from TPC-DS benchmark.

```
(slice0)    Executor memory: 322K bytes.  Peak memory: 11783K bytes.  Vmem reserved: 12288K bytes.
   (slice1)    Executor memory: 8375K bytes avg x 3 workers, 8375K bytes max (seg0).
  Peak memory: 9488K bytes avg x 3 workers, 9488K bytes max (seg0).  Vmem reserved: 9216K bytes avg x 3 workers, 9216K bytes max (seg0).
 Total memory used across slices: 40247K  bytes
 Statement statistics:
   Memory used: 128000K bytes
```

Signed-off-by: Sambitesh Dash <sdash@pivotal.io>

### CODE REVIEW

File: src/backend/cdb/cdbexplain.c [hunk 1]
```diff
@@ -2097,6 +2099,8 @@ cdbexplain_showExecStatsEnd(struct PlannedStmt *stmt,
 								 "  Peak memory: %s%s.",
 								 maxbuf,
 								 seg);
+				slice_memory = atoi(maxbuf);
```
Comment (hsyuan) [inline src/backend/cdb/cdbexplain.c]: we have pg_atoi, use this

File: src/backend/cdb/cdbexplain.c [hunk 2]
```diff
@@ -2108,6 +2112,8 @@ cdbexplain_showExecStatsEnd(struct PlannedStmt *stmt,
 								 ss->memory_accounting_global_peak.vcnt,
 								 maxbuf,
 								 segbuf);
+				slice_memory = (atoi(maxbuf) * ss->memory_accounting_global_peak.vcnt);
+				total_memory_across_slices += slice_memory;
```
Comment (hsyuan) [inline src/backend/cdb/cdbexplain.c]: this is duplicate, can we extract this out?

File: src/backend/cdb/cdbexplain.c [hunk 3]
```diff
@@ -2164,6 +2170,11 @@ cdbexplain_showExecStatsEnd(struct PlannedStmt *stmt,
 		appendStringInfoChar(str, '\n');
 	}
 
+	if (total_memory_across_slices > 0)
+	{
+		appendStringInfo(str, "Total memory used across slices: %dK  bytes \n", total_memory_across_slices);
```
Comment (hsyuan) [inline src/backend/cdb/cdbexplain.c]: still `%d` for `long`?

Comment (hsyuan) [reply src/backend/cdb/cdbexplain.c]: extra space before `bytes`

File: src/backend/cdb/cdbexplain.c (hunk 1 above)
Comment (sambitesh) [reply src/backend/cdb/cdbexplain.c]: Sure thanks.

File: src/backend/cdb/cdbexplain.c (hunk 2 above)
Comment (sambitesh) [reply src/backend/cdb/cdbexplain.c]: Will do it

**Pull request merged**